bindings/src/main/java/io/pravega/storage/filesystem/FileSystemStorage.java

```diff
@@ -303,28 +303,29 @@ private Void doWrite(SegmentHandle handle, long offset, InputStream data, int le
             throw new StreamSegmentSealedException(handle.getSegmentName());
         }
 
-        long fileSize = path.toFile().length();
-        if (fileSize < offset) {
-            throw new BadOffsetException(handle.getSegmentName(), fileSize, offset);
-        } else {
-            long totalBytesWritten = 0;
-            try (FileChannel channel = FileChannel.open(path, StandardOpenOption.WRITE)) {
-                // Wrap the input data into a ReadableByteChannel, but do not close it. Doing so will result in closing
-                // the underlying InputStream, which is not desirable if it is to be reused.
-                ReadableByteChannel sourceChannel = Channels.newChannel(data);
-                while (length != 0) {
-                    long bytesWritten = channel.transferFrom(sourceChannel, offset, length);
-                    assert bytesWritten > 0 : "Unable to make any progress transferring data.";
-                    offset += bytesWritten;
-                    totalBytesWritten += bytesWritten;
-                    length -= bytesWritten;
-                }
+        long totalBytesWritten = 0;
+        try (FileChannel channel = FileChannel.open(path, StandardOpenOption.WRITE)) {
+            long fileSize = channel.size();
+            if (fileSize != offset) {
```

Comment:
The condition in the old code to throw bad offset is that the filesize is smaller than than the offset, but here it is changing to different (`!=`), why?

Reply:
To make it consistent with how other storage (HDFS and S3) are checking the same conditions. @andreipaduroiu suggested the change and I agree with the suggestion.